teraserver/python/tests/modules/FlaskModule/API/user/test_UserQueryDeviceTypes.py

```diff
@@ -0,0 +1,183 @@
+        # Create a new device type ID = 5
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # ID = 6
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device_1'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # ID = 7
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device_2'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # Delete without params
+        params = {}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 500)
+
+        # Deleting the new device type
+        params = {'device_type_key': 'new_device'}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # Deleting the new device 1 type
+        params = {'id_device_type': 6}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # Try deleting again an unexisting Device type
+        params = {'id_device_type': 6}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 500)
+
+        # Try deleting 2 devices at once
+        params = {'id_device_type': 7, 'device_type_key': 'capteur'}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 501)
+
+        # Deleting the same device with 2 parameters
+        params = {'id_device_type': 7, 'device_type_key': 'new_device_2'}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+    def _delete_with_http_auth(self, username, password, payload=None):
+        from requests import delete
+        if payload is None:
+            payload = {}
+        url = self._make_url(self.host, self.port, self.test_endpoint)
+        return delete(url=url, verify=False, auth=(username, password), params=payload)
```

Comment:
This should be moved in 'BaseAPITest.py' to be reused, if needed, by other tests.

Reply:
Should it replace the original _delete_with_http_auth or be renamed with _delete_with_https_auth_payload?

Reply:
It should be renamed, since the original _delete_with_http_auth is used in all the other tests, and cover the case for most DELETE APIs where we only manage the id to delete, not other parameters.

Reply:
renamed _delete_with_http_auth_plus